src/cmstats.c

```diff
@@ -49,6 +50,26 @@ s_duplicator (const void *self)
     return (void*) fty_proto_dup ((fty_proto_t*) self);
 }
 
+static double
+safe_atof (const char *str)
+{
+    char *endptr = (char *) zmalloc (sizeof (char));
+    errno = 0;
+    double ret = strtod (str, &endptr);
+    // str does not contain number
+    if (ret == 0 && endptr == str) {
```

Comment:
Comparison `ret == 0.0` seems cleaner, to stress (to compiler too) the floatiness of this zero :)